lang/test/org/partiql/lang/eval/EvaluatingCompilerOrderByTests.kt

```diff
@@ -0,0 +1,105 @@
+            ),
+            // should order by supplierId_nulls nulls first
+            EvaluatorTestCase(
+                "SELECT productId, supplierId_nulls FROM products_sparse ORDER BY supplierId_nulls NULLS FIRST, productId",
+                "[{'productId': 4, 'supplierId_nulls': NULL}, {'productId': 5, 'supplierId_nulls': NULL}, {'productId': 8, 'supplierId_nulls': NULL}, {'productId': 9, 'supplierId_nulls': NULL}, {'productId': 10, 'supplierId_nulls': NULL}, {'productId': 1, 'supplierId_nulls': 10}, {'productId': 2, 'supplierId_nulls': 10}, {'productId': 3, 'supplierId_nulls': 10}, {'productId': 6, 'supplierId_nulls': 11}, {'productId': 7, 'supplierId_nulls': 11}]"
+            ),
+            // should order by nulls last as default for supplierId_nulls asc
+            EvaluatorTestCase(
+                "SELECT productId, supplierId_nulls FROM products_sparse ORDER BY supplierId_nulls ASC, productId",
+                "[{'productId': 1, 'supplierId_nulls': 10}, {'productId': 2, 'supplierId_nulls': 10}, {'productId': 3, 'supplierId_nulls': 10}, {'productId': 6, 'supplierId_nulls': 11}, {'productId': 7, 'supplierId_nulls': 11}, {'productId': 4, 'supplierId_nulls': NULL}, {'productId': 5, 'supplierId_nulls': NULL}, {'productId': 8, 'supplierId_nulls': NULL}, {'productId': 9, 'supplierId_nulls': NULL}, {'productId': 10, 'supplierId_nulls': NULL}]"
+            ),
+            // should order by nulls first as default for supplierId_nulls desc
+            EvaluatorTestCase(
+                "SELECT productId, supplierId_nulls FROM products_sparse ORDER BY supplierId_nulls DESC, productId",
+                "[{'productId': 4, 'supplierId_nulls': NULL}, {'productId': 5, 'supplierId_nulls': NULL}, {'productId': 8, 'supplierId_nulls': NULL}, {'productId': 9, 'supplierId_nulls': NULL}, {'productId': 10, 'supplierId_nulls': NULL}, {'productId': 6, 'supplierId_nulls': 11}, {'productId': 7, 'supplierId_nulls': 11}, {'productId': 1, 'supplierId_nulls': 10}, {'productId': 2, 'supplierId_nulls': 10}, {'productId': 3, 'supplierId_nulls': 10}]"
+            ),
+            // should group and order by asc sellerId
+            EvaluatorTestCase(
+                "SELECT sellerId FROM orders GROUP BY sellerId ORDER BY sellerId ASC",
+                "[{'sellerId': 1}, {'sellerId': 2}]"
+            ),
+            // should group and order by desc sellerId
+            EvaluatorTestCase(
+                "SELECT sellerId FROM orders GROUP BY sellerId ORDER BY sellerId DESC",
+                "[{'sellerId': 2}, {'sellerId': 1}]"
+            )
+        )
+    }
```

Comment:
This is a good start for the evaluation tests. I'll discuss further with the team about what level of coverage seems reasonable as `NaturalExprValueComparatorsTest` covers a lot of edge cases. Can consider adding more tests with
- `MISSING`, `NULL` and `MISSING`
- different data types (not just numbers)

Reply:
I have added more test cases which covers different data types and edge cases